ctapipe_io_magic/__init__.py

```diff
@@ -405,6 +425,51 @@ def get_run_info_from_name(file_name):
 
         return run_number, is_mc, telescope, datalevel
 
+    def check_files(self):
+        """Check the the input files contain the needed trees."""
+
+        needed_trees = ["RunHeaders", "Events"]
+        num_files = len(self.files_)
+
+        if (
+            num_files == 1
+            and "Drive" not in self.files_[0].keys(cycle=False)
+            and "OriginalMC" not in self.files_[0].keys(cycle=False)
+        ):
+            logger.error("Cannot proceed without Drive information for a single file.")
+            return False
+
+        if (
+            num_files == 1
+            and "Trigger" not in self.files_[0].keys(cycle=False)
+            and "OriginalMC" not in self.files_[0].keys(cycle=False)
+        ):
+            logger.error(
+                "Cannot proceed without Trigger information for a single file."
+            )
+            return False
+
+        for rootf in self.files_:
+            for tree in needed_trees:
+                if tree not in rootf.keys(cycle=False):
+                    logger.warning(
+                        f"File {rootf.file_path} does not have the tree {tree}."
+                    )
+                    if tree == "RunHeaders" or tree == "Events":
+                        logger.error(
+                            f"Cannot proceed without RunHeaders or Events tree. "
+                            f"File {rootf.file_path} will be excluded."
+                        )
+                        self.files_.remove(rootf)
+                        self.excluded_files_.append(rootf)
+                        break
+
+        if len(self.excluded_files_) == num_files:
+            logger.error("All input files were excluded.")
+            return False
+
+        return True
```

Comment:
Events and Runheaders tree should be always in the file, unless it is really broken, so how about fail the test if a single file is missing one of those trees?
Alternatively we could also return here as a second argument the number of broken files, and depending on a config file switch fail with one or with all broken files.

Reply:
Ok, I now stop execution if at least only one file is not valid. No need to return a number of broken files, the log will tell which files are affected.